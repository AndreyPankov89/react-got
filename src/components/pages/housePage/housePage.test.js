import React from 'react';
import HousePage from './housePage';
import {shallow} from 'enzyme';

describe('Testing <HousePage/>', () => {
    const item = shallow(<HousePage/>);
    describe('testing shap & state', () =>{
        it('HousePage have rendered correctly', () => {
            expect(item).toMatchSnapshot();
        })
    
        it("HousePage state item is empty object", () =>{
            expect(item.state().selectedHouse).toBeNull();
        })
        it("HousePage state error is false", () =>{
            expect(item.state().error).toBeFalsy();
        })
    });
    describe('Handlers tests', () => {
        it('testing onitemSelected', () => {
            item.instance().onItemSelected(1);
            expect(item.state().selectedHouse).toEqual(1)
        });
        it('testing onError', () => {
            item.instance().onError({message:"2"});
            expect(item.state().error).toBeTruthy();
            expect(item.state().loading).toBeFalsy();
            expect(item.state().errorCode).toBe('2')
        });
        
    })
})