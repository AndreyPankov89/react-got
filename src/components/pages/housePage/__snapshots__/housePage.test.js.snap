// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <HousePage/> testing shap & state HousePage have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HouserPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "left": <ItemList
        getData={[Function]}
        onError={[Function]}
        onItemSelected={[Function]}
        renderItem={[Function]}
      />,
      "right": <ItemDetails
        getItem={[Function]}
        itemId={null}
        what="house"
      >
        <Field
          field="region"
          label="Region"
        />
        <Field
          field="words"
          label="Words"
        />
        <Field
          field="titles"
          label="Titles"
        />
        <Field
          field="overlord"
          label="Overlord"
        />
        <Field
          field="ancestralWeapons"
          label="Ancestral Weapone"
        />
      </ItemDetails>,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "left": <ItemList
          getData={[Function]}
          onError={[Function]}
          onItemSelected={[Function]}
          renderItem={[Function]}
        />,
        "right": <ItemDetails
          getItem={[Function]}
          itemId={null}
          what="house"
        >
          <Field
            field="region"
            label="Region"
          />
          <Field
            field="words"
            label="Words"
          />
          <Field
            field="titles"
            label="Titles"
          />
          <Field
            field="overlord"
            label="Overlord"
          />
          <Field
            field="ancestralWeapons"
            label="Ancestral Weapone"
          />
        </ItemDetails>,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
